### DISCUSSION TITLE
Should there be a Deprecated Plugin

### DISCUSSION THREAD

**Discussion by lumbar527** [endless-sky/endless-sky#8964] [Polls]
**Deprecated Plugin**
It would be a plugin of the `_deprecated` files plus some images.

**Why Use a Plugin?**
Because at some point, the `_deprecated` folder will be removed and some saves will be broken.

**Links**
My repo:
https://github.com/lumbar527/es-compatability-plugin
A zip copy:
[es-compatability plugin.zip](https://github.com/endless-sky/endless-sky/files/11995175/es-compatability.plugin.zip)
Related discussion:
https://github.com/endless-sky/endless-sky/pull/8908#issuecomment-1593399149

**Comment by EjoThims** [comment]
I would much, much rather transition to this sooner than later, and far prefer this option to both wholly erasing it all at 1.0 -and- to continuing to bloat standard game files to support content which no longer has a place in the actual standard game.

**Reply by quyykk** [reply]
We can always do this for 1.0, but I don't think it's a good idea to do it right now because this will effectively break backwards compatibility for players who don't install that plugin (and yes installing a plugin is easy, but it's one more step to do which in practice means that a lot of people won't do).

**Comment by TomGoodIdea** [comment]
I agree with EjoThims and I think patch missions should be moved to the plugin too.

Also I prefer naming it "ES Compatibility Plugin".

**Reply by EjoThims** [reply]
I really like this idea as well.

I could see finding and extracting all the patch missions as a project in itself, though, so wouldn't mind moving forward with "ES Compatibility Plug-in" starting with just the deprecated stuff and then doing those in an update.

This would then also separate the different extractions from the ES files into (at least) two different PRs, which may be best anyway.

**Reply by lumbar527** [reply]
I've renamed the plugin and added patch missions.

**Comment by mOctave** [comment]
I think it should be moved into a plugin, yes, but there should also be a system in-game to explicitly tell you when you have deprecated stuff as opposed to just broken stuff, and directing you to where you can get the plugin. Otherwise, we might as well just purge all the old data.

**Reply by TomGoodIdea** [reply]
We would still need to have a deprecated things list in game files (including patch missions' trigger conditions) and probably we would also need to implement a way to read the list and display the message.
So I think it would be better to say "if you started your game before 0.9.X, you may need to install the plugin".

**Reply by mOctave** [reply]
Maybe, but there's probably going to be new deprecated stuff fairly often until we hit 1.0.0. We'd need some sort of way to distinguish between new changes and old in what the version number is. Even just a list of keywords that have been deprecated would reduce data file bloat, although it wouldn't be perfect.
